### ISSUE TITLE
Feature request : having more control on searching

### ISSUE THREAD

**Issue by eseb63** [aaFn/Bookmark-search-plus-2#94]
thanks for this great successor of alice75's extension

some suggestions for having better control on searching : 
- search for exact word or expression
- search in title only, address only, or both 
- search globally or locally (the selected folder)

thanks for your attention

**Comment by aaFn** [comment]
Hello @eseb63 , this https://github.com/aaFn/Bookmark-search-plus-2/issues/90#issuecomment-475911532
should fit your desires . let me know if you have comments there ...

**Comment by eseb63** [comment]
hi @aaFn ,
your suggestions 1 (search field) and 2 (search scope) fit my requests 2 and 3.

i see in the api  definition 

> "Search terms are space-delimited and may be enclosed in quotes to allow multiple-word phrases to be searched against."

 does that syntax handle exact matching (my request 1)?

and if so, did you ask the Firefox team why it's not working? 
exact matching is sometimes very helpful : it would be nice to fix this issue.

**Comment by aaFn** [comment]
No, I didn't ask. Anybody can do, just open a bug on bugzilla :-)

**Comment by eseb63** [comment]
recently back at home ; as an ex developer, i feel your response quite surprising : you encounter a bug which breaks a feature and annoy probably some users but you don't try to fix it ?  investigate and ask the Firefox team is the developer's job, not the user's one : only you knows exactly the using context of this api in your code ?

**Comment by aaFn** [comment]
Hello @eseb63 , this is what I do for annoying bugs indeed.
This one frankly is very specific ... users who know how to interpret and how to use such things as

> Search terms are space-delimited and may be enclosed in quotes to allow multiple-word phrases to be searched against.

are quite rare, and skilled enough to open themselves a bug if they feel it more annoying than me :-)

On this one, I do not have the time to invest to characterize the bug, to verify I understand properly the sentence and use things properly, and then to follow up on that one.

**Comment by eseb63** [comment]
Hello @aaFn,
the api definition looks quite classic for searching ?

Searching exact expressions, for a searching tool like yours, is not a minor feature, but a classic and basic one : without it, the search may in some cases return  many inaccurate results (i had the problem several times). So for me, it would justify to invest some time for that, before invest certainly much more time for complex features as, for example, regular expressions that you seemed to acccept though  (#90).

you may find more interesting to implement regexp but its clear that it will be used by few users, because using regexp is not user friendly. Searching exact expression is much more simple, so would be used by much more users, so think to them ?

and please stop saying the user can investigate in your place, it's a bad joke :)

**Comment by aaFn** [comment]
To be honest, the space delimited feature of the API works well. And it works exactly the same between BSP2 and the native bookmark sidebar.

Enclosing in quotes .. how many users are really going to do that ? Roughly the same than the ones ok for using a more complex syntax ..
By the way, quotes are not working either on the native bookmark sidebrt (or let e know if you find a way to make it work ..)

Have you already open and followed a bug with FF ? I believe average time for correction is between 6 month and 2 years, and you can find bugs which are open there since much much longer ...

Using quotes is not a crucial feature, it would have been a good bonus if it were working .. but it is not.
so I do not advertise it. I mentioned it to you because you were querying on things like that, but you will be the only one.

Since I will look at something more complete anyway, I do not want to spend time on that one, And I do not want to advertise on something which is not working, and which may get corrected in some years (or even, they may simply remove that part of text from the spec ..).

In short, I learnt to work around by myself when something is needed rather than to wait for it.
As you say, search is a core feature, so I will work on getting something sooner.


Once again, I leave it to others if they want to fight windmills. I do not.

Cheers, aaFn.

**Comment by eseb63** [comment]
You can't honestly pretend that using quotes will concern the same users than using regexp : only developers and a few others will use regexp (very few) ; enclosing an expression with quotes is a common syntax in search engines like Google, so many of your users  (most of them) know this syntax and could use this feature if they need it.

you recently opened a bug on Firefox (autofocus in search bar) : is this feature crucial ? not more than enclosing with quotes i mean. So, having to wait seems only problematic when you need it to refuse a request.

in #31  also you gave bad reasons to refuse the request.

So, trying to convince you seems useless as you decided to refuse  arbitrarily, whatever arguments and as you prefer sometimes using bad faith or ignore what we say (like for my last response in #31 ).

Listening to the user needs is part of developer's job : you seem to forget it.

so, keep making as you want, and only what you want : trying to convince you is also fighting windmills...

**Comment by aaFn** [comment]
Hey, you do not need to be aggressive.

As I said to you, when there is a workaround, this is how I learnt to address it on FF. Therefore I won't open a bug on it, I will act in my code as part of the regexp feature, and will let others create the bug as they need on FF. Since you appear to have a lot of energy to spend on others, you can be the one .. 
On autofocus, there is no workaround possible .. this is why I opened it.

Last, if I were not listening, I wouldn't be here with you, or wouldn't have had all the modifications I made since the beginning on user demands.
Simply, as I already explained to you, this is a matter of time and effort on my spare time . and I am not a slave. Thank you for your understanding.

AaFn.

**Comment by eseb63** [comment]
you listen sometimes, but in this case (and #31 ) you stay deaf to everything, only to stay on your position and use bad faith if necessary ; some examples :

-  bad faith when you pretend that users knowing to use expressions enclosed in quotes are the same that regexp users ; you have near 2500 users, regexp users will probably represent less than 10% of them, the others maybe 80% or more (they probably use it when searching with Google) ; 

- bad faith when you pretend a user can investigate on this bug (or misunderstanding) without knowing the development platform, and how  **YOU** use thi api in **YOUR** code

- bad faith when you say that "collapsing all" is only justified if "expanding all" is justified too, although you probably know that the honest interpretation is the inverse one, and that "collapsing all" may also be considered alone

so maybe you felt me aggressive, but i hate bad faith ...

**Comment by aaFn** [comment]
Regexp searching is available in 2.0.65 for more precise searches as you need, as well as other features to limit to only titles or URLs, or to search only within a folder.

Closing this issue, thank you.
